CommonCPPCTest.cmake

```diff
@@ -50,9 +50,9 @@ macro(common_add_cpp_test NAME FILE)
   endif()
 
   if(NAME MATCHES "^perf.*")
-    list(APPEND ALL_CPP_PERF_TESTS ${TEST_NAME})
+    list(APPEND ALL_CPP_PERF_TESTS ctest_${TEST_NAME})
   else()
-    list(APPEND ALL_CPP_TESTS ${TEST_NAME})
+    list(APPEND ALL_CPP_TESTS ctest_${TEST_NAME})
```

Comment:
ctest_${TEST_NAME} target is appended to the list before being defined, which is a bit confusing. I would suggest moving it after add_custom_target(ctest_${TEST_NAME}).